### PR TITLE
Add support for notify_keyspace_events

### PR DESCRIPTION
**Pull request by dcharleston**

### CODE REVIEW

File: templates/redis_port.conf.erb [hunk 1]
```diff
@@ -420,6 +420,54 @@ slowlog-log-slower-than <%= @redis_slowlog_log_slower_than %>
 # You can reclaim memory used by the slow log with SLOWLOG RESET.
 slowlog-max-len <%= @redis_slowlog_max_len %>
 
+<% if @redis_notify_keyspace_events %>
```
Comment (SegFaultAX) [inline templates/redis_port.conf.erb]: I suspect this is going to introduce newlines in the config which, while not having any impact on the interpretation of the config file, will cause *some* changes and possibly cycle any redis servers that still have the service managed.

File: manifests/instance.pp [hunk 2]
```diff
@@ -147,6 +148,10 @@
     }
   }
 
+  if !empty($redis_notify_keyspace_events) and versioncmp($version, 2.8.0) >= 0 {
```
Comment (SegFaultAX) [inline manifests/instance.pp]: Did you test this?

**Comment by SegFaultAX** [comment]
This looks awesome and really close to exactly what we want! I really urge you to test this module very carefully, especially against servers provisioned with the current module. Can you create a cluster in vagrant, upgrade the module, and see what impact the new config options have?

Provisionally, :shipit: 💩 :trollface:

**Comment by anshul-s** [comment]
Looks good to me. So this means we will need to create a new role for these PIN nodes(Premium Inventory) and then create new hiera keys for the role while using the same master?

File: templates/redis_port.conf.erb (hunk 1 above)
Comment (dcharleston) [inline templates/redis_port.conf.erb]: I changed %> to -%> so no newlines are created when this config isn't used

File: manifests/instance.pp (hunk 2 above)
Comment (dcharleston) [inline manifests/instance.pp]: This is working, the original condition was wrong but its been updated

**Comment by dcharleston** [comment]
This is tested.  There is no affect to the config if redis_notify_keyspace_events is not supplied, even if puppet is managing the config.

**Comment by SegFaultAX** [comment]
:shipit: 💩 :trollface:

**Pull request merged**